aptos-move/framework/aptos-framework/sources/account_recovery.move

```diff
@@ -0,0 +1,148 @@
+module aptos_framework::account_recovery {
+
+    const ERECOVERY_NOT_SET: u64 = 1;
+    const ERECOVERY_ALREADY_SET: u64 = 2;
+    const EAUI: u64 = 2;
+
+    use std::signer;
+    use aptos_std::table::{Table, add};
+    use aptos_framework::account;
+    use aptos_framework::system_addresses;
+    use aptos_framework::timestamp;
+    use aptos_std::table;
+    use std::vector;
+    use std::option::Option;
+    use std::option;
+    use aptos_framework::resource_account;
+    use aptos_framework::account::SignerCapability;
+
+    /// Account recovery struct
+    struct AccountRecovery has key, store, drop {
+        authorized_addresses: vector<address>,
+        required_delay_seconds: u64,
+        account_recovery_init: Option<AccountRecoveryInitData>,
+        // TODO add recovery capability here.
+    }
+
+    // Reverse lookup map
+    struct AccountRecoveryReverseLookup has key, store {
+        authorized_to_recovery: Table<address, vector<address>>,
+    }
+
+    struct AccountRecoveryInitData has key, store {
+        recovery_seq_number: u64,
+        recovery_initiation_ts: u64,
+    }
+
+    struct ModuleData has key {
+        // Storing the signer capability here, so the module can programmatically sign for transactions
+        signer_cap: SignerCapability,
+    }
+
+
+    fun initialize(signer: &signer) acquires AccountRecoveryReverseLookup {
```

Comment:
this function should probably be called `init_module` - `init_module` will be automatically called during the module publishing process. if we don't call it during the module publishing process, we essentially lose the resource signer and won't be able to do 
```
        let resource_signer_cap = resource_account::retrieve_resource_account_cap(signer, @source_addr);
```